recipes/init.rb

```diff
@@ -54,3 +54,41 @@
     end
   end
 end
+
+include_recipe 'krb5' if hadoop_kerberos?
+
+princ =
+  if node['cdap']['cdap_site'].key?('cdap.master.kerberos.principal')
+    node['cdap']['cdap_site']['cdap.master.kerberos.principal']
+  else
+    "#{hdfs_user}/_HOST"
```

Comment:
so we default to the ``yarn`` principal?  seems a little wierd?  should we just default to ``cdap`` since we are creating principals?